packages/ckbtc/src/minter.canister.ts

```diff
@@ -192,19 +194,33 @@ export class CkBTCMinterCanister extends Canister<CkBTCMinterService> {
     }).retrieve_btc_status({ block_index: transactionId });
 
   /**
-   * Returns the status of all BTC withdrawals for the user's main account.
+   * Returns the status of all BTC withdrawals for an account.
    *
    * @param {boolean} certified query or update call
+   * @param {MinterAccount} account an optional account to retrieve the statuses. If not provided, statuses for the caller are retrieved.
    * @returns {Promise<RetrieveBtcStatusV2WithId[]>} The statuses of the BTC retrieval requests.
    */
   retrieveBtcStatusV2ByAccount = async ({
+    account,
     certified,
-  }: {
-    certified: boolean;
-  }): Promise<RetrieveBtcStatusV2WithId[]> => {
-    const statuses = await this.caller({
+  }: RetrieveBtcStatusV2ByAccountParams): Promise<
+    RetrieveBtcStatusV2WithId[]
+  > => {
+    const { retrieve_btc_status_v2_by_account } = this.caller({
       certified,
-    }).retrieve_btc_status_v2_by_account([]);
+    });
+
+    const statuses = await retrieve_btc_status_v2_by_account(
+      isNullish(account)
+        ? []
+        : [
+            {
+              owner: account.owner,
+              subaccount: toNullable(account.subaccount),
+            },
+          ],
```

Comment:
This is what `toNullable` is for, right? (wrapping a nullable in an array for candid optional)

Reply:
Not sure what you mean here?

Reply:
I thought you could just do `toNullable(account)` but I realize now that the subaccount needs to be dealt with.
Though this is a quite common conversion we do. Do we not have a function to do it already?
Otherwise you could do this but that doesn't make much difference although it's still more concise:
```
    const statuses = await retrieve_btc_status_v2_by_account(
      toNullable(account && {
              owner: account.owner,
              subaccount: toNullable(account.subaccount),
            }),
```